src/backend/joanie/core/forms.py

```diff
@@ -0,0 +1,67 @@
+"""
+Core application forms declaration
+"""
+from django import forms
+from django.contrib.admin import widgets
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+from django.utils.translation import gettext_lazy as _
+
+from joanie.core import models
+
+
+class ProductCourseRelationAdminForm(forms.ModelForm):
+    """
+    The admin form for the ProductCourseRelation model.
+
+    This is a customized form in order to filter the list of course runs to
+    their which relies on the course instance.
+    """
+
+    course_runs = forms.ModelMultipleChoiceField(
+        queryset=models.CourseRun.objects.none(),
+        widget=widgets.FilteredSelectMultiple(
+            models.CourseRun._meta.verbose_name_plural, is_stacked=False
+        ),
+        required=False,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        instance = kwargs.get("instance")
+        if instance is not None:
+            queryset = models.CourseRun.objects.filter(
+                Q(course=instance.course) | Q(pk__in=instance.course_runs.all())
+            )
+            self.fields["course_runs"].queryset = queryset
+
+
+class CourseRunAdminForm(forms.ModelForm):
+    """
+    The admin form for the CourseRun model.
+
+    It implements a clean_course method to prevent integrity error. In fact update
+    course_run has to be forbid if the course run is joined to a product or an order.
+    """
+
+    def clean_course(self):
```

Comment:
I was wondering where to put this verification.

I mainly choose to put this here because currently this form is the only way to update a course run and I'm able to know which fiels has changed here.

From the model, I have to implement a more complex logic to know which field has been changed. But ok I'll implement it on the model as later we will need to be able to update course run through api.